retrieval/targetmanager.go

```diff
@@ -33,7 +33,10 @@ import (
 //
 // The TargetProvider does not have to guarantee that an actual change happened.
 // It does guarantee that it sends the new TargetGroup whenever a change happens.
-// On startup it sends all TargetGroups it can see.
+//
+// Sources() is guaranteed to be called before each call to Run(). On a call to Run()
+// implementing types must send a valid target group for each of the sources they
```

Comment:
What if don't? "Must" might be too strong. What happens if they send through a `TargetGroup` that is not related to a source?